tests/schemas/test_schema.py

```diff
@@ -107,3 +107,56 @@ class Aclass:
     msg = "Can't instantiate abstract class BaseSchemaDefinition with abstract methods get_rendered_fields, render"
 
     assert msg == str(excinfo.value)
+
+
+def test_get_enum_type_map(user_advance_dataclass_with_enum, color_enum, user_type_enum):
+    assert user_advance_dataclass_with_enum._get_enum_type_map() == {
+        "favorite_colors": color_enum,
+        "user_type": user_type_enum,
+    }
+
+
+def test_get_enum_type_map_with_unions(user_advance_dataclass_with_union_enum, color_enum, user_type_enum):
+    assert user_advance_dataclass_with_union_enum._get_enum_type_map() == {
+        "favorite_colors": color_enum,
+        "user_type": user_type_enum,
+    }
+
+
+def test_get_enum_type_map_with_sub_record(user_advance_dataclass_with_sub_record_and_enum, color_enum, user_type_enum):
```

Comment:
Consider parametrizing those tests, as they are very similar.
```python
@pytest.mark.parametrize(
     "schema_cls",
     [user_advance_dataclass_with_enum, ...]. # Note: unfortunately, one cannot use fixtures here. Concrete classes have to be passed here.
)
def test_get_enum_type_map(schema_cls: AvroModel):
    assert schema_cls._get_enum_type_map() == {
        "favorite_colors": color_enum,
        "user_type": user_type_enum,
    }

Reply:
@pawelrubin It's not clear to me how to parametrize these tests without pulling the classes out of the fixtures. Is there a cleaner option you're aware of?

Reply:
On second thought, it could make things more complicated. Too bad that **pylint** does not support using fixtures in `pytest.mark.parametrize`. However, you could try using [this](https://github.com/pytest-dev/pytest/issues/349#issuecomment-189370273) approach. It's a bit hacky though. Up to you.

Reply:
I think I'll opt to leave these tests as is then if we're all okay with that.